### PR TITLE
[openrr-plugin] Add openrr-plugin

### LINKED ISSUE #75: arci plugin system
We want make it a plugin!

Is it possible to do that for async trait?

### PR DESCRIPTION
**Pull request by taiki-e**
Closes #75

## Examples

- [plugin definition](https://github.com/openrr/openrr/blob/taiki-e/plugin/openrr-plugin/examples/plugin/src/lib.rs)
- [plugin loader](https://github.com/openrr/openrr/blob/taiki-e/plugin/openrr-plugin/examples/load_plugin.rs)

### API Design

- [x] Decide on a unit of the plugin.
- [x] Decide what arguments to use when initializing the plugin.
- [x] Decide naming convention of proxy types (R* is confusing.)

### Implementation

- [x] arci::JointTrajectoryClient
- [x] arci::Speaker
- [x] arci::MoveBase
- [x] arci::Navigation
- [x] arci::Localization
- [x] arci::TransformResolver
- [x] arci::Gamepad

### Other

- [x] docs
- [x] tests
- [x] benches

## Future plan

- support openrr-plugin in openrr-apps #312 
- tests more

### CODE REVIEW

File: openrr-plugin/tests/proxy.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+};
+use assert_approx_eq::assert_approx_eq;
+use nalgebra::Vector2;
+use openrr_plugin::{RLocalization, RMoveBase, RNavigation, RSpeaker};
+
+// TODO: move to arci?
+#[derive(Debug, Default)]
+struct DummySpeaker {
+    message: Mutex<String>,
+}
```
Comment (taiki-e) [inline openrr-plugin/tests/proxy.rs:18]: Opened #311

File: openrr-plugin/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+    ///
+    /// NOTE: This is *not* a unique identifier.
+    fn name(&self) -> String;
+
+    /// Creates a new instance of [`arci::JointTrajectoryClient`] with the specified arguments.
+    fn new_joint_trajectory_client(
+        &self,
+        args: String,
+    ) -> Option<Box<dyn StaticJointTrajectoryClient>> {
+        drop(args);
```
Comment (OTL) [inline openrr-plugin/src/lib.rs:109]: Just a question as a rust beginner.
Why we need drop?

File: openrr-plugin/examples/plugin/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+[package]
+name = "openrr-plugin-example"
+version = "0.1.0"
+authors = ["Taiki Endo <te316e89@gmail.com>"]
```
Comment (OTL) [inline openrr-plugin/examples/plugin/Cargo.toml:4]: Please update mail

File: openrr-plugin/src/lib.rs (hunk 2 above)
Comment (taiki-e) [reply openrr-plugin/src/lib.rs:109]: - Without it, `args` will be warned as an unused variable.
- There are two major ways to silence that warning, `#[allow(..)]` and leading `_`, but the former is not nice to look at and the latter is not nice to look when displayed in the documentation. Also, rust-analyzer generates method templates based on the trait definitions, so the user has to start writing by removing the leading `_`.

As a workaround for those problems, `drop` or `let _ =` sometimes be used in the default method for public traits.

- https://docs.rs/syn/1.0.72/src/syn/parse.rs.html#580-583
- https://github.com/serde-rs/serde/blob/1094e2d334f12705957f579f49adfc6118a1de1f/serde/src/ser/mod.rs#L1948
- https://github.com/tokio-rs/valuable/blob/c3efa86b4ac919fc14fe642eb5dfa55916e46e6f/valuable/src/visit.rs#L6

Comment (taiki-e) [reply openrr-plugin/src/lib.rs]: Switched from `drop` to `let _ =` because that seems to be a more widely used pattern for that purpose.

Comment (OTL) [reply openrr-plugin/src/lib.rs]: I see, thanks!

**Review by OTL — APPROVED** [review]

**Pull request merged**